scripts/models.py

```diff
@@ -67,37 +67,57 @@ def linear_nn():
         data_path = Path('../data')
 
     predictor = LinearNetwork(layer_sizes=[100], data_folder=data_path)
-    predictor.train()
+    predictor.train(num_epochs=50, early_stopping=5)
     predictor.evaluate(save_preds=True)
 
-    # mostly to test it works
-    test_arrays_loader = DataLoader(data_path=data_path, batch_file_size=1,
-                                    shuffle_data=False, mode='test', to_tensor=True)
-    key, val = list(next(iter(test_arrays_loader)).items())[0]
-
-    explain_hist, explain_add = predictor.explain([val.x.historical[:3], val.x.pred_months[:3]])
-    print(explain_hist.shape)
-    np.save('shap_linear_network_hist.npy', explain_hist)
-    np.save('shap_linear_network_add.npy', explain_add)
-
-    np.save('shap_x_linear_network_hist.npy', val.x.historical[:3])
-    np.save('shap_x_linear_network_add.npy', val.x.pred_months[:3])
-
-    with open('variables_linear_network.txt', 'w') as f:
-        f.write(str(val.x_vars))
+    # The code below is commented out because of a bug in the shap deep Explainer which
+    # prevents it from working. It has been fixed in master, but not deployed yet:
+    # https://github.com/slundberg/shap/pull/684
+
+    # test_arrays_loader = DataLoader(data_path=data_path, batch_file_size=1,
+    #                                 shuffle_data=False, mode='test', to_tensor=True)
+    # key, val = list(next(iter(test_arrays_loader)).items())[0]
+    #
+    # explain_hist, explain_add = predictor.explain([val.x.historical[:3], val.x.pred_months[:3]])
+    # print(explain_hist.shape)
+    # np.save('shap_linear_network_hist.npy', explain_hist)
+    # np.save('shap_linear_network_add.npy', explain_add)
+    #
+    # np.save('shap_x_linear_network_hist.npy', val.x.historical[:3])
+    # np.save('shap_x_linear_network_add.npy', val.x.pred_months[:3])
+    #
+    # with open('variables_linear_network.txt', 'w') as f:
+    #     f.write(str(val.x_vars))
+    #
+    # # plot the variables
+    # with (data_path / 'features/normalizing_dict.pkl').open('rb') as f:
+    #     normalizing_dict = pickle.load(f)
+    #
+    # for variable in val.x_vars:
+    #     plt.clf()
+    #     plot_shap_values(val.x[0].numpy(), explain_hist[0], val.x_vars, normalizing_dict, variable,
+    #                      normalize_shap_plots=True, show=False)
+    #     plt.savefig(f'{variable}_linear_network.png', dpi=300, bbox_inches='tight')
+
+
+def rnn():
+    # if the working directory is alread ml_drought don't need ../data
+    if Path('.').absolute().as_posix().split('/')[-1] == 'ml_drought':
+        data_path = Path('data')
+    else:
+        data_path = Path('../data')
 
-    # plot the variables
-    with (data_path / 'features/normalizing_dict.pkl').open('rb') as f:
-        normalizing_dict = pickle.load(f)
+    predictor = RecurrentNetwork(hidden_size=128,
```

Comment:
So this will create a RNN with 
[ input_size -> 100 ] * 128 layers?

Reply:
There is only one RNN layer, but its hidden state has 128 dimensions (so the output of each RNN cell will have shape `[batch_size, 128]`.

Then, the model takes the last hidden state, and puts it through a linear model with layer sizes `[128, 1]`.